ecosystem/indexer/src/token_processor.rs

```diff
@@ -148,22 +149,23 @@ fn update_token_ownership(
     conn: &PgPoolConnection,
     token_id: String,
     txn: &UserTransaction,
-    amount_update: i64,
+    amount_update: bigdecimal::BigDecimal,
 ) {
     let ownership = Ownership::new(
         token_id,
         txn.sender.clone(),
-        amount_update,
+        ensure_above_zero(amount_update.clone()),
```

Comment:
this doesn't has to be above zero. This is a delta change. when withdraw happens, amount_update is negative used for deducting tokens from existing amount

Reply:
this is NOT a delta change: this is an absolute update :-)

Reply:
I make this value negative here  by multiplying with -1 https://github.com/aptos-labs/aptos-core/blob/5e4a93b968ba285702207f80c1b1692347df65e8/ecosystem/indexer/src/token_processor.rs#L226 when withdraw events happens.

Reply:
yes, because the previous type was i64. I highly recommend running the indexer on this branch, and removing the `ensure_above_zero`, as it'll explode around `163000` :-)

Reply:
or even simpler: https://dbfiddle.uk/?rdbms=postgres_13&fiddle=8827d0671962240e1071e6156d3fda04

Reply:
Ok, I see your point. 
1. after changing the type, it doesn't work as the value cannot be negative
2. previously, it works as it is i64 and can represent negative delta. 

Now, I need to figure out a different way to deduct the `amount_update` from ownership_amount. I can do it in a followup PR

Reply:
The way I have it now works as you intended :-) the delta update is okay as a delta, as long as the final value isn’t below zero

Reply:
you are right. it works as intended. Miss read the code last night.